Sources/SwiftGen/Logs.swift

```diff
@@ -47,7 +47,7 @@ func logMessage(_ level: LogLevel, _ string: CustomStringConvertible) {
     case .info:
       fputs(ANSIColor.green.format("\(string)\n"), stdout)
     case .warning:
-      fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
+      fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stdout)
     case .error:
       fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
```

Comment:
Should we use `stdout` for errors here if we detect that the context is Xcode (like I'm sure there's some env var like `XCODE_VERSION` defined we can check)? Or do errors get logged properly by Xcode even in `stderr` and it's only for warnings that it didn't work (would be a strange asymmetry from Xcode… but 🤷 )

iirc we decided on that (stdout for warnings, stderr for errors) because that's how SwiftLint also does it, but still it's worth checking…

Reply:
Not 100% sure about this one. We have a check for "dumb" terminals, but would that match too much? (we currently use it to decide whether to output in color)

Reply:
I'm confused now.

Added a shell script build phase to swiftgen to test
```sh
echo "swiftgen: warning: to stdout"
>&2 echo "swiftgen: warning: to stderr"
echo "swiftgen: error: to stdout"
>&2 echo "swiftgen: error: to stderr"
```
<img width="493" alt="image" src="https://user-images.githubusercontent.com/216089/85565819-c56a0800-b62f-11ea-9f7a-52596b8af4d7.png">

But it prints all the warnings correctly in all cases…
Or was this an issue with a specific and older version of Xcode which has been fixed now?
<img width="636" alt="image" src="https://user-images.githubusercontent.com/216089/85565732-b3886500-b62f-11ea-9c19-a203868a83f2.png">

Reply:
Possibly...

Reply:
I also just remembered that the initial reason for our logs to be printed in stderr is that swiftgen allows to print the generated content to stdout (`swiftgen xcassets --templateName swift5 Assets.xcassets` will print to stdout since there's no `--output`). See also `OutputDestination`. How should we handle those cases so that when printing generated content to stdout the output is not mixed up with the log messages then?

Maybe one more reason to conditionally print to stderr vs stdout depending on being on Xcode? 🤔

Reply:
Well, seeing your tests above, it doesn't matter if we print to `stderr` or not. I think that for now we should revert this change, and ask the OP in #547 if they still have the issue.

Reply:
Yeah, I agree. Let's revert and keep it to stderr for now, ping the OP (and close the issue if no response in a couple of months or so)